### PR TITLE
feat: convert to TypeScript

### PR DESCRIPTION
**Pull request by JustinBeckwith**
This is a first draft.  Needs work to get the system tests passing :)

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    this.db = options.dataset;
+    if (!this.db) {
+      throw new Error('No dataset provided to Firestore Session.');
+    }
+    this.kind = options.kind || 'Session';
+  }
+
+  get =
+      (sid: string,
+       callback: (err?: Error|null, session?: Express.SessionData|null) =>
```
Comment (fhinkel) [inline src/index.ts]: Why not the pattern where you pass in the session at construction?

**Comment by fhinkel** [comment]
@JustinBeckwith are you seeing this through or should I take over? Thanks!

**Comment by JustinBeckwith** [comment]
A take over would be great if you have cycles.

**Comment by fhinkel** [comment]
I don't, but I'll make cycles 🌀

**Comment by fhinkel** [comment]
I fixed the system tests 🌀 🌀 

Tests are failing because we need to run them with a different project (Firestore and Datastore cannot be activated in the same project): 
``` ifError got unwanted exception: 9 FAILED_PRECONDITION: The Cloud Firestore API is not available for Cloud Datastore projects.```

**Pull request merged**